object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/MustCallInvokedChecker.java

```diff
@@ -338,44 +357,141 @@ private void handleAssignment(AssignmentNode node, Set<LocalVarWithTree> newDefs
+          newDefs.add(newSetContainingLatestAssignmentPair);
+        }
       }
 
       // If the rhs is an ObjectCreationNode, or a MethodInvocationNode, then it adds
       // the AssignmentNode to the newDefs.
       if ((rhs instanceof ObjectCreationNode)
           || (rhs instanceof MethodInvocationNode
               && !hasNotOwningReturnType((MethodInvocationNode) rhs))) {
-        newDefs.add(
-            new LocalVarWithTree(new LocalVariable((LocalVariableNode) lhs), node.getTree()));
+        if (typeFactory.hasMustCallChoice(rhs.getTree())) {
+          LocalVariableNode n = getLocalPassedAsMustCallChoiceParam(rhs);
+          if (n != null && isVarInDefs(newDefs, n)) {
+            ImmutableSet<LocalVarWithTree> immutableSet =
+                getSetContainingAssignmentTreeOfVar(newDefs, n);
+            ImmutableSet<LocalVarWithTree> newImmutableSet =
+                FluentIterable.from(immutableSet)
+                    .append(
+                        new LocalVarWithTree(
+                            new LocalVariable((LocalVariableNode) lhs), node.getTree()))
+                    .toSet();
+            newDefs.remove(immutableSet);
+            newDefs.add(newImmutableSet);
+          }
+        } else {
+          LocalVarWithTree lhsLocalVarWithTreeNew =
+              new LocalVarWithTree(new LocalVariable((LocalVariableNode) lhs), node.getTree());
+          newDefs.add(ImmutableSet.of(lhsLocalVarWithTreeNew));
+        }
       }
 
       // Ownership Transfer
       if (rhs instanceof LocalVariableNode && isVarInDefs(newDefs, (LocalVariableNode) rhs)) {
         // If the rhs is a LocalVariableNode that exists in the newDefs (Note that if a
         // localVariableNode exists in the newDefs it means it isn't assigned to a null
         // literals), then it adds the localVariableNode to the newDefs
-        newDefs.add(
-            new LocalVarWithTree(new LocalVariable((LocalVariableNode) lhs), node.getTree()));
-        newDefs.remove(getAssignmentTreeOfVar(newDefs, (LocalVariableNode) rhs));
+        ImmutableSet<LocalVarWithTree> setContainingRhs =
+            getSetContainingAssignmentTreeOfVar(newDefs, (LocalVariableNode) rhs);
+        LocalVarWithTree lhsLocalVarWithTreeNew =
+            new LocalVarWithTree(new LocalVariable((LocalVariableNode) lhs), node.getTree());
+        newDefs.add(ImmutableSet.of(lhsLocalVarWithTreeNew));
+        newDefs.remove(setContainingRhs);
+      }
+    }
+    return newDefs;
+  }
+
+  /**
+   * This method tries to find a local variable passed as a @MustCallChoice parameter. In the base
+   * case, if {@code node} is a local variable, it just gets returned. Otherwise, if node is a call
+   * (or a call wrapped in a cast), the code finds the parameter passed in the @MustCallChoice
+   * position, and recurses on that parameter.
+   *
+   * @param node
+   * @return {@code node} iff {@code node} represents a local variable, otherwise null
```

Comment:
This documentation on the `@return` is not correct

Reply:
https://github.com/kelloggm/object-construction-checker/pull/242/commits/73a5d0d6f607756067b47b9b65bd846ff1a9c71d